src/view.c

```diff
@@ -671,43 +671,25 @@ view_handle_destroy(struct view *view)
 			view->server->cycle_view, LAB_CYCLE_DIR_BACKWARD);
 
 		/* If we cycled back to ourselves, then we have no windows.
-		 * just remove it and close the osd for good. */
-		if (view->server->cycle_view == view) {
+		 * just remove it and close the OSD for good. */
+		if (view->server->cycle_view == view || !view->server->cycle_view) {
 			view->server->cycle_view = NULL;
```

Comment:
Why? We don't have an OSD.

Reply:
It is within the inner branch so the view that is being destroyed was the selected one and `desktop_cycle_view()` returned `NULL` (no focusable view found).

So instead of showing an empty OSD we are closing it, even if the modifier is still pressed at this point.